metricflow/test/fixtures/id_fixtures.py

```diff
@@ -10,13 +11,30 @@
 from metricflow.dag.prefix_id import PrefixIdGenerator
 
 
+@dataclass(frozen=True)
 class IdNumberSpace:
     """Defines the numbering of IDs when setting up tests and test fixtures."""
 
-    # Before each test run.
-    TEST_START = 0
-    # When setting up the ConsistentIdObjectRepository
-    CONSISTENT_ID_REPOSITORY = 10000
+    start_value: int
+
+    @staticmethod
+    def for_test_start() -> IdNumberSpace:
+        """Before each test run."""
+        return IdNumberSpace(0)
+
+    @staticmethod
+    def for_block(block_index: int, block_size: int = 2000, first_block_offset: int = 10000) -> IdNumberSpace:
+        """Used to define fixed-size blocks of ID number spaces.
+
+        This is useful for creating unique / non-overlapping IDs for different groups of objects.
+        """
+        if not block_index >= 0:
+            raise RuntimeError(f"block_index should be >= 0. Got: {block_index}")
+        if not block_size > 1:
+            raise RuntimeError(f"block_size should be > 1. Got: {block_size}")
+        if not first_block_offset >= 0:
+            raise RuntimeError(f"first_block_offset should be >= 0. Got: {first_block_offset}")
+        return IdNumberSpace(first_block_offset + block_size * block_index)
```

Comment:
In future, can we please avoid mixing large mechanical changes like renames with structural or functional changes (or even other renames)?

It's hard to convey just how much harder it is to review this commit as a result, and in other contexts this kind of thing leads to preventable incidents because reviewers get confused about the intersection of these changes and fail to recognize when some small thing is out of place in the mass of mechanical updates.

Reply:
Yeah, I've been meaning to avoid such cases as I do get that it makes for a more challenging review.